Firestore/core/src/firebase/firestore/remote/serializer.cc

```diff
@@ -697,22 +695,114 @@ google_firestore_v1_DocumentMask Serializer::EncodeDocumentMask(
 
   int i = 0;
   for (const FieldPath& path : mask) {
-    result.field_paths[i] = EncodeString(path.CanonicalString());
+    result.field_paths[i] = EncodeFieldPath(path);
     i++;
   }
 
   return result;
 }
 
 /* static */
-model::FieldMask Serializer::DecodeDocumentMask(
+FieldMask Serializer::DecodeFieldMask(
     const google_firestore_v1_DocumentMask& mask) {
   std::set<FieldPath> fields;
   for (size_t i = 0; i < mask.field_paths_count; i++) {
-    auto path = DecodeString(mask.field_paths[i]);
-    fields.insert(FieldPath::FromServerFormat(path));
+    fields.insert(DecodeFieldPath(mask.field_paths[i]));
+  }
+  return FieldMask(std::move(fields));
+}
+
+/* static */
+google_firestore_v1_DocumentTransform_FieldTransform
+Serializer::EncodeFieldTransform(const FieldTransform& field_transform) {
+  using Type = TransformOperation::Type;
+
+  google_firestore_v1_DocumentTransform_FieldTransform proto{};
+  proto.field_path = EncodeFieldPath(field_transform.path());
+
+  switch (field_transform.transformation().type()) {
+    case Type::ServerTimestamp:
+      proto.which_transform_type =
+          google_firestore_v1_DocumentTransform_FieldTransform_set_to_server_value_tag;  // NOLINT
```

Comment:
Question: what does the linter complain about, line length?

Reply:
Yes. Clang-format can't help get it under 80 columns.

Reply:
Ah, I'd expect it to be smart enough to figure out that a single "word" cannot be split and should be exempt.